enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -157,429 +172,528 @@ def make_map(
-    y_max = pingrid.tile_top_mercator(ty, tz)
-    y_min = pingrid.tile_top_mercator(ty + 1, tz)
-
-    if (
+        return timeseries_plot
+
+    def select_season(target_season):
+        if target_season == 'MAM':
+            season_str = 'Mar-May'
+        if target_season == 'JJA':
+            season_str = 'Jun-Aug'
+        if target_season == 'SON':
+            season_str = 'Sep-Nov'
+        if target_season == 'DJF':
+            season_str = 'Dec-Feb'    
+        return season_str 
+
+
+    @APP.callback(
+        Output("map_title","children"),
+        Input("data_choice","value"),
+        Input("target_year","value"),
+        Input("target_season","value"),
+    )
+    def write_map_title(data_choice, target_year, target_season):
+        season_str = select_season(target_season)
+        map_title = (
+            f"{config['map_text'][data_choice]['menu_label']} "
+            f"for {season_str} in {str(target_year)}"
+        )
+        return map_title
+
+
+    @FLASK.route(f"{TILE_PFX}/<int:tz>/<int:tx>/<int:ty>")
+    def cropSuit_layers(tz, tx, ty):
+        parse_arg = pingrid.parse_arg
+        data_choice = parse_arg("data_choice")
+        target_season = parse_arg("target_season")
+        target_year = parse_arg("target_year", float)  
+        data_choice = parse_arg("data_choice")
+        min_wet_days = parse_arg("min_wet_days", int)
+        wet_day_def = parse_arg("wet_day_def", float)
+        lower_wet_threshold = parse_arg("lower_wet_threshold", int)
+        upper_wet_threshold = parse_arg("upper_wet_threshold", int)
+        maximum_temp = parse_arg("maximum_temp", float)
+        minimum_temp = parse_arg("minimum_temp", float)
+        temp_range = parse_arg("temp_range", float)
+
+        # Reads daily data
+        zarr_path_rr = GLOBAL_CONFIG["datasets"]["daily"]["vars"]["precip"][1]
+        if zarr_path_rr is None:
+            zarr_path_rr = GLOBAL_CONFIG["datasets"]["daily"]["vars"]["precip"][0]
+        rr_mrg = calc.read_zarr_data(Path(
+            f'{GLOBAL_CONFIG["datasets"]["daily"]["zarr_path"]}{zarr_path_rr}'
+        ))[GLOBAL_CONFIG["datasets"]["daily"]["vars"]["precip"][2]]
+        zarr_path_tmin = GLOBAL_CONFIG["datasets"]["daily"]["vars"]["tmin"][1]
+        if zarr_path_tmin is None:
+            zarr_path_tmin = GLOBAL_CONFIG["datasets"]["daily"]["vars"]["tmin"][0]
+        tmin_mrg = calc.read_zarr_data(Path(
+            f'{GLOBAL_CONFIG["datasets"]["daily"]["zarr_path"]}{zarr_path_tmin}'
+        ))[GLOBAL_CONFIG["datasets"]["daily"]["vars"]["tmin"][2]]
+        zarr_path_tmax = GLOBAL_CONFIG["datasets"]["daily"]["vars"]["tmax"][1]
+        if zarr_path_tmax is None:
+            zarr_path_tmax = GLOBAL_CONFIG["datasets"]["daily"]["vars"]["tmax"][0]
+        tmax_mrg = calc.read_zarr_data(Path(
+            f'{GLOBAL_CONFIG["datasets"]["daily"]["zarr_path"]}{zarr_path_tmax}'
+        ))[GLOBAL_CONFIG["datasets"]["daily"]["vars"]["tmax"][2]]
+
```

Comment:
Lots of hard-to-read boilerplate whenever we open a dataset. Do you have a plan for how to improve this, or just an intention to work on it? Feel free to bounce ideas off me.

Reply:
no plan but intention as of now. Will plan when I start the exercise to split more cleanly data reading away from the apps.